app/graphql/types/profile.rb

```diff
@@ -22,9 +24,7 @@ class Profile < Types::BaseObject
     def rules(args = {})
       selected_columns = args[:lookahead].selections.map(&:name) &
                          ::Rule.column_names.map(&:to_sym)
-      rules = object.rules.select(selected_columns << :id).where(
-        id: RuleResult.selected.pluck(:rule_id)
```

Comment:
This was really the crux of the problem. `RuleResult.selected` will return all rules with a valid result.. no matter the host. So the following could happen:

1. Host A checks in with policy PCI which contains 1025 rules, of which 90 are valid (not `notselected`, `notapplicable` etc..)
2. System Details page looks good because there are no other hosts, and this `RuleResult.selected` just returns the valid rules for the current host.
3. Host B checks in with policy STIG-DISA from the same SSG as Host A. Policy contains 1025 rules, of which 90 are valid. This set of valid rules is DIFFERENT to the one in point 1, it may have some overlap.
4. Now both Host B and Host A system details rules table show "120 rules" instead of "90". Why? Because the rules returned by this query contain all rules with a valid result no matter the policy .